pkg/arvo/sys/hoon.hoon

```diff
@@ -10619,6 +10637,10 @@
         {$zpmc *}
       =+  vos=$(gol %noun, gen q.gen)       ::  XX validate!
       [(nice (cell (play p.gen) p.vos)) (cell (play(sut dox) p.gen) q.vos)]
+    ::
+        {$zpgl *}
+      ::  XX is this right?
```

Comment:
It looks right

Reply:
I actually don't think it is quite right for the same reason that `!>` is currently mishandled in +mull.  Both of those should give a mull-bonk error if the actual type is not exactly the formal type.  We can't generate the correct `!<` nest check or `!>` +burp type polymorphically, since it depends on the actual type.

Reply:
So, a check that sut and dox mutually nest for both `!;` and `!<`? In a sense, these runes implicitly and silently dry out the core, whereas such a check would force the caller to do so.